src/main/java/software/amazon/awssdk/trustedidentitypropagation/TrustedIdentityPropagationPlugin.java

```diff
@@ -103,6 +105,14 @@ private TrustedIdentityPropagationPlugin(Builder builder) {
         this.webTokenProvider = builder.webTokenProvider;
         this.applicationRoleArn = builder.applicationRoleArn;
 
+        if (builder.ssoOidcClient == null || builder.stsClient == null) {
+            Validate.notNull(builder.applicationRoleArn,
+                    "Either both of ssoOidcClient and stsClient OR the application role must be provided.");
+            Validate.isFalse(builder.accessRoleArn.equals(builder.applicationRoleArn),
+                    "Access Role and Application Role can not be same.");
```

Comment:
Nit: Should we provide customers more context as to why Access and Application roles can not be the same?